var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -154,6 +154,9 @@ class QtBase(QtPackage):
 
     # Qt6 requires newer compilers: see https://github.com/spack/spack/issues/34418
     conflicts("%gcc@:7")
+    # The oldest compiler for Qt 6.5 is GCC 9: https://doc.qt.io/qt-6.5/supported-platforms.html
```

Comment:
As discussed [here](https://github.com/spack/spack/issues/34418#issuecomment-1344331389) as well, this link is not minimum *working*, but minimum *supported*. What is the change that breaks gcc-8 compatibility and was it introduced with 6.5 or could it have been in 6.4 already too?

Reply:
6.4 compiles fine with GCC 8, while 6.5 does not. The reason that I noticed is that references to routines from std::filesystem cannot be resolved, as GCC 8 requires linking to -lstdc++fs for this. Neither did I explore, if there are further incompatibilities, nor investigate, how complicated it is to patch Qt 6.5.